packages/lib-classifier/src/plugins/drawingTools/models/marks/Polygon/Polygon.js

```diff
@@ -0,0 +1,120 @@
+import { getParentOfType, types } from 'mobx-state-tree'
+import { Polygon as PolygonComponent } from '@plugins/drawingTools/components/'
+import { Mark } from '@plugins/drawingTools/models/marks'
+import { PolygonTool } from '@plugins/drawingTools/models/tools'
+import roundCoordinates from '@helpers/roundCoordinates'
+
+const MINIMUM_POINTS = 3
+
+const singleCoord = types.model({
+  x: types.maybe(types.number),
+  y: types.maybe(types.number)
+})
+
+const PolygonModel = types
+  .model('PolygonModel', {
+    points: types.array(singleCoord)
+  })
+  .volatile(() => ({
+    guideLineX: types.maybe(types.number),
+    guideLineY: types.maybe(types.number)
+  }))
+  .views((self) => ({
+    get coords() {
+      return {
+        x: self.points[0]?.x,
+        y: self.points[0]?.y
+      }
+    },
+
+    deleteButtonPosition(scale) {
+      const BUFFER = 16
+      const x = self.points[0].x - BUFFER / scale
+      const y = self.points[0].y - BUFFER / scale
+      return { x, y }
+    },
+
+    get isValid() {
+      return self.points.length >= MINIMUM_POINTS
+    },
+
+    get tool() {
+      return getParentOfType(self, PolygonTool)
+    },
+
+    get initialPoint() {
+      const [firstCoord] = self.points
+      if (!firstCoord) {
+        return null
+      }
+      return firstCoord
+    },
+
+    get lastPoint() {
+      const lastCoord = [...self.points].pop()
+      if (!lastCoord) {
+        return null
+      }
+      return lastCoord
+    },
+
+    // path = string. Used to create the <polyline>
+    get path() {
+      const [firstCoord, ...otherCoords] = self.points
+      if (!firstCoord) {
+        return ''
+      }
+      let path = `${firstCoord.x},${firstCoord.y} `
+      otherCoords.forEach(({ x, y }) => {
+        path = path + `${x},${y} `
+      })
+      return path
+    },
+
+    get toolComponent() {
+      return PolygonComponent
+    }
+  }))
+  .actions((self) => ({
+    initialDrag() {},
```

Comment:
Good to see this has been manually added, as per the [drawing tools documentation/intended design patterns](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/plugins/drawingTools/README.md) we were talking about. Nice! 👍